frontend/packages/pipelines-plugin/src/components/catalog/providers/useTasksProvider.tsx

```diff
@@ -0,0 +1,91 @@
+import * as React from 'react';
+import { TFunction } from 'i18next';
+import * as _ from 'lodash';
+import { useTranslation } from 'react-i18next';
+import { CatalogItem, ExtensionHook } from '@console/dynamic-plugin-sdk';
+import { getImageForIconClass } from '@console/internal/components/catalog/catalog-item-icon';
+import { useK8sWatchResources } from '@console/internal/components/utils/k8s-watch-hook';
+import { referenceForModel } from '@console/internal/module/k8s';
+import { TaskModel, ClusterTaskModel } from '../../../models';
+import { TaskKind } from '../../../types';
+import { TektonTaskAnnotation, TektonTaskLabel } from '../../pipelines/const';
+
+const normalizeTektonTasks = (TektonTasks: TaskKind[], t: TFunction): CatalogItem<TaskKind>[] => {
+  const normalizedTektonTasks: CatalogItem<TaskKind>[] = _.reduce(
+    TektonTasks,
+    (acc, task) => {
+      const { uid, name, annotations = {}, creationTimestamp, labels = {} } = task.metadata;
+      const { description } = task?.spec;
+      const tags = (annotations[TektonTaskAnnotation.tags] || '').split(/\s*,\s*/);
+      const provider = labels[TektonTaskLabel.providerType];
+      const iconClass = 'icon-build';
+      const imgUrl = getImageForIconClass(iconClass);
+
+      const normalizedTektonTask: CatalogItem<TaskKind> = {
+        uid,
+        type: 'Tekton',
+        name,
+        description,
+        provider,
+        tags,
+        creationTimestamp,
+        icon: {
+          url: imgUrl,
+          class: iconClass,
+        },
+        cta: {
+          label: t('pipelines-plugin~Add'),
+          callback: () => {},
+        },
+        data: task,
+      };
+      acc.push(normalizedTektonTask);
+      return acc;
+    },
+    [],
+  );
+
+  return normalizedTektonTasks;
+};
+
+const useTasks: ExtensionHook<CatalogItem[]> = ({
+  namespace,
+}): [CatalogItem[], boolean, string] => {
+  const { t } = useTranslation();
+
+  const { namespacedTasks, clusterTasks } = useK8sWatchResources<{
+    namespacedTasks: TaskKind[];
+    clusterTasks: TaskKind[];
+  }>({
+    namespacedTasks: {
+      kind: referenceForModel(TaskModel),
+      isList: true,
+      namespace,
+    },
+    clusterTasks: {
+      kind: referenceForModel(ClusterTaskModel),
+      isList: true,
+      namespaced: false,
+    },
+  });
```

Comment:
No access to Formik state from here, eh? Not in the tree so we can't just consume the forms resources?

Reply:
It is in the tree, I have updated to use it.